corehq/apps/api/resources/v0_5.py

```diff
@@ -157,7 +158,14 @@ def obj_create(self, bundle, request=None, **kwargs):
             self._update(bundle)
             bundle.obj.save()
         except Exception:
-            bundle.obj.retire()
+            if bundle.obj._id:
```

Comment:
I prefer avoiding exceptions if they can be replaced by similarly complex if...else logic, since the control flow is more predictable